File: corpCode/xml_to_db.py
import sqlite3
import xml.etree.ElementTree as ET

xml_path = "corpCode/CORPCODE.xml"
db_path = "corpCode/corpcode.db"

conn = sqlite3.connect(db_path)
cur = conn.cursor()
cur.execute("""
CREATE TABLE IF NOT EXISTS corp (
    corp_code TEXT PRIMARY KEY,
    corp_name TEXT,
    stock_code TEXT,
    modify_date TEXT
)
""")
conn.commit()

context = ET.iterparse(xml_path, events=("end",))
for event, elem in context:
    if elem.tag == "list":
        corp_code = elem.findtext("corp_code")
        corp_name = elem.findtext("corp_name")
        stock_code = elem.findtext("stock_code")
        modify_date = elem.findtext("modify_date")
        cur.execute(
            "INSERT OR REPLACE INTO corp VALUES (?, ?, ?, ?)",
            (corp_code, corp_name, stock_code, modify_date)
        )
        elem.clear()
conn.commit()
conn.close()
print("DB 변환 완료!") 

File: backend/xml_to_db.js
const fs = require('fs');
const Database = require('better-sqlite3');
const xml2js = require('xml2js');

const xmlPath = './corpCode/CORPCODE.xml';
const dbPath = './corpCode/corpcode.db';

const db = new Database(dbPath);
db.prepare(`CREATE TABLE IF NOT EXISTS corp (
  corp_code TEXT PRIMARY KEY,
  corp_name TEXT,
  stock_code TEXT,
  modify_date TEXT
)`).run();

fs.readFile(xmlPath, (err, data) => {
  if (err) throw err;
  xml2js.parseString(data, (err, result) => {
    if (err) throw err;
    const list = result.result.list;
    const insert = db.prepare("INSERT OR REPLACE INTO corp VALUES (?, ?, ?, ?)");
    const insertMany = db.transaction((items) => {
      for (const item of items) {
        insert.run(
          item.corp_code[0],
          item.corp_name[0],
          item.stock_code[0],
          item.modify_date[0]
        );
      }
    });
    insertMany(list);
    db.close();
    console.log('DB 변환 완료!');
  });
}); 